add-permissions.js

```diff
@@ -13,6 +13,18 @@ module.exports = class AwsAddLambdaAccountPermissions {
     this.hooks = {
       'package:createDeploymentArtifacts': () => this.beforeDeploy(),
     };
+    this.serverless.configSchemaHandler.defineFunctionProperties(this.provider, {
+      properties: {
+        allowAccess: { type: 'array' },
+      },
+    });
+   this.serverless.configSchemaHandler.defineProvider(this.provider, {
```

Comment:
`defineProvider` defined a whole new provider, based on the description. Though, there is no function to define a new property for the existing provider. Not sure if using `this.provider` is OK.